CMakeLists.txt

```diff
@@ -16,24 +16,66 @@ project(
     LANGUAGES CXX
 )
 
+#---- Options.
+option(KOKKOSUTILS_ENABLE_TESTS "Enable testing" ON)
+option(KOKKOSUTILS_ENABLE_DOC   "Enable documentation" ON)
+
 #---- C++ standard that we need (at least).
-set(CMAKE_CXX_STANDARD          20   CACHE BOOL "" FORCE)
-set(CMAKE_CXX_STANDARD_REQUIRED True CACHE BOOL "" FORCE)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 #---- Find Kokkos.
-find_package(
-    Kokkos
-    ${KOKKOS_VERSION}
-    REQUIRED
-)
+if(NOT TARGET Kokkos::kokkoscore)
+    find_package(
+        Kokkos
+        ${KOKKOS_VERSION}
+        REQUIRED
+    )
+endif()
 
 #---- Find Google Test.
-find_package(
-    GTest
-    ${GOOGLETEST_VERSION}
-    CONFIG
-    REQUIRED
-)
+if(NOT TARGET GTest::gtest)
+    find_package(
+        GTest
+        ${GOOGLETEST_VERSION}
+        CONFIG
+        REQUIRED
+    )
+    endif()
 
 #---- Documentation.
-add_subdirectory(docs)
+if(KOKKOSUTILS_ENABLE_DOC)
+    #--- Find Doxygen.
+    if(NOT TARGET Doxygen::doxygen)
+        find_package(
+            Doxygen
+            ${DOXYGEN_VERSION}
+            REQUIRED
+        )
+    endif()
+
+    add_subdirectory(docs)
+endif()
+
+#---- Build the Kokkos::Utils library.
+add_library(KokkosUtils INTERFACE)
+add_library(Kokkos::Utils ALIAS KokkosUtils)
+
+target_sources(
+    KokkosUtils
+    INTERFACE
+        FILE_SET concepts
+        TYPE HEADERS
+        BASE_DIRS include
+        FILES include/concepts/View.hpp
+)
+
+target_include_directories(KokkosUtils INTERFACE "include/kokkos-utils")
```

Comment:
There's an issue left here. We'd like to use includes like `concepts/View.hpp` here, but `Kokkos-utils/concepts/View.hpp` in downstream projects.